ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/BaseApplicationEngine.kt

```diff
@@ -33,11 +35,20 @@ public abstract class BaseApplicationEngine(
             it.sendPipeline.merge(pipeline.sendPipeline)
             it.receivePipeline.installDefaultTransformations()
             it.sendPipeline.installDefaultTransformations()
+            it.installDefaultInterceptors()
         }
         environment.monitor.subscribe(ApplicationStarted) {
             environment.connectors.forEach {
                 environment.log.info("Responding at ${it.type.name.toLowerCase()}://${it.host}:${it.port}")
             }
         }
     }
+
+    private fun Application.installDefaultInterceptors() {
```

Comment:
We had a lot of tests relying on isHandled (or isUnhandled). I wonder, why didn't they stop working after this change?

Reply:
I updated `isHandled` so it is false for 404 responses here https://github.com/ktorio/ktor/pull/2115/files#diff-a3f83b71ecd565b0a1089db3230bccfabdaf774b35a2101deba655f840245bebR75

Reply:
Also, some tests were updated to compare status with 404